plugins/animate/animate.cpp

```diff
@@ -247,6 +255,19 @@ class wayfire_animation : public wf::singleton_plugin_t<animation_global_cleanup
     void set_animation(wayfire_view view,
         wf_animation_type type, int duration)
     {
+        if (type == ANIMATION_TYPE_UNMAP)
+        {
+            view->take_ref();
+            view->take_snapshot();
+        }
+
+        if (view->has_data(animate_custom_data_id))
+        {
+            view->get_data<animation_hook_base>(
+                animate_custom_data_id)->reverse(type);
```

Comment:
What happens if these are different animations, for ex. we had unminimize animation running, then the view is unmapped and it should start a fire animation?

Reply:
You mean something like this?


https://user-images.githubusercontent.com/1450125/127919341-dcaebd7e-c042-4139-b539-0cb2df35301b.mp4

Reply:
Yep, this is how it should be.